server/taskflows/hpccloud/taskflow/nwchem.py

```diff
@@ -0,0 +1,290 @@
+
+    finally:
+        os.remove(path)
+
+@cumulus.taskflow.task
+def setup_input(task, *args, **kwargs):
+    input_folder_id = kwargs['input']['folder']['id']
+    geometry_file_id = kwargs['input']['geometryFile']['id']
+    kwargs['geometryFileId'] = geometry_file_id
+
+    number_of_procs = kwargs.get('numberOfSlots')
+    if not number_of_procs:
+        number_of_procs = kwargs.get('numberOfNodes')
+
+    if not number_of_procs:
+        size = parse('cluster.config.launch.params.node_instance_count').find(kwargs)
+        if size:
+            number_of_procs = size[0].value + 1
+        else:
+            raise Exception('Unable to extract number of nodes in cluster')
+
+
+    if not number_of_procs:
+        raise Exception('Unable to determine number of mpi processes to run.')
+
+    number_of_procs = int(number_of_procs)
+    kwargs['numberOfProcs']  = number_of_procs
+
+    client = _create_girder_client(
+        task.taskflow.girder_api_url, task.taskflow.girder_token)
+
+    # Get the geometry file metadata to see if we need to import
+    geometry_file = client.getResource('file/%s' % geometry_file_id)
+    kwargs['geometryFilename'] = geometry_file['name']
+
+    update_config_file(task, client, *args, **kwargs)
+
+    ini_file_id = kwargs['input']['iniFile']['id']
+    ini_file = client.getResource('file/%s' % ini_file_id)
+    kwargs['iniFilename'] = ini_file['name']
+
+    create_job.delay(*args, **kwargs)
+
+
+@cumulus.taskflow.task
+def create_job(task, *args, **kwargs):
+    task.logger.info('Taskflow %s' % task.taskflow.id)
+    task.taskflow.logger.info('Create NWChem job.')
+    input_folder_id = kwargs['input']['folder']['id']
+
+    # TODO: setup command to run with mpi
+    body = {
+        'name': 'nwchem_run',
+        'commands': [
+            "nwchem input/%s" % (kwargs['iniFilename'])
+        ],
+        'input': [
+            {
+              'folderId': input_folder_id,
+              'path': 'input'
+            }
+        ],
+        'output': [],
+        'params': {
+            'numberOfSlots': kwargs['numberOfProcs']
+        }
+    }
+
+    client = _create_girder_client(
+                task.taskflow.girder_api_url, task.taskflow.girder_token)
+
+    job = client.post('jobs', data=json.dumps(body))
+
+    task.taskflow.set_metadata('jobs', [job])
+
+    submit.delay(job, *args, **kwargs)
+
+@cumulus.taskflow.task
+def submit(task, job, *args, **kwargs):
+    task.taskflow.logger.info('Submitting job to cluster.')
+    girder_token = task.taskflow.girder_token
+    cluster = kwargs.pop('cluster')
+    task.taskflow.set_metadata('cluster', cluster)
+
+    # Now download and submit job to the cluster
+    task.logger.info('Uploading input files to cluster.')
+    download_job_input_folders(cluster, job, log_write_url=None,
+                        girder_token=girder_token, submit=False)
+    task.logger.info('Uploading complete.')
```

Comment:
Since we have a client/server/cluster model rather than just client/server it can get confusing. Maybe we should say moving or transferring files to cluster?